### PR TITLE
[feature](vectorized) Support MemoryScratchSink on vectorized engine

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.54 seconds
 load time: 654 seconds
 storage size: 17122987107 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230104122557_clickbench_pr_73476.html

File: be/src/util/arrow/block_convertor.cpp [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+#include "gutil/strings/substitute.h"
+#include "runtime/descriptor_helper.h"
+#include "runtime/descriptors.h"
+#include "runtime/large_int_value.h"
+#include "util/arrow/utils.h"
+#include "util/types.h"
+
+namespace doris {
+
+// Convert RowBatch to an Arrow::Array
```
Comment (morningman) [inline be/src/util/arrow/block_convertor.cpp:48]:
```suggestion
// Convert Block to an Arrow::Array
```

File: be/src/util/arrow/block_convertor.cpp [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                             << "to arrow type";
+                return arrow::Status::TypeError("unsupported column type");
+            }
+            }
+        }
+        return builder.Finish(&_arrays[_cur_field_idx]);
+    }
+
+    // process doris DecimalV2
+    arrow::Status Visit(const arrow::Decimal128Type& type) override {
```
Comment (HappenLee) [inline be/src/util/arrow/block_convertor.cpp:133]: Add dispose decimalv3 and datetime v2？

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**